Revit_Core_Engine/Query/DrawingArea.cs

```diff
@@ -0,0 +1,101 @@
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Autodesk.Revit.DB;
+using BH.Engine.Geometry;
+using BH.oM.Adapters.Revit.Settings;
+using BH.oM.Base.Attributes;
+using BH.oM.Geometry;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Compute the outline of Title Block drawing area.")]
+        [Input("document", "The current Revit document to be processed.")]
+        [Input("titleBlock", "Title Block symbol to get the drawing area outline from.")]
+        [Output("outline", "The Title Block's drawing area.")]
+        public static Outline DrawingArea(this Document document, FamilySymbol titleBlock)
+        {
+            Document familyDoc = document.EditFamily(titleBlock.Family);
+            List<DetailLine> lines = new FilteredElementCollector(familyDoc).OfCategory(Autodesk.Revit.DB.BuiltInCategory.OST_Lines).WhereElementIsNotElementType().Where(x => x is DetailLine).Cast<DetailLine>().ToList();
+
+            var compositeGeom = new CompositeGeometry();
+
+            foreach (DetailLine dLine in lines)
+            {
+                var bhomLine = dLine.GeometryCurve.IFromRevit() as BH.oM.Geometry.Line;
+                compositeGeom.Elements.Add(bhomLine);
+            }
+
+            BH.oM.Geometry.Point centrePoint = compositeGeom.Bounds().Centre();
+
+            var upLine = BH.Engine.Geometry.Create.Line(centrePoint, centrePoint + Vector.YAxis * 10);
```

Comment:
Why do you multiply by factor 10? Is that because the sheet will never be larger than 10x10m?

Reply:
Yes 👍

I changed it to `double.MaxValue`

Reply:
> Yes 👍
> 
> I changed it to `double.MaxValue`

Moved it back to `10` because `double.MaxValue` doesn't work.

Reply:
Yeah, that's what I thought when saw your reply about max value 😉 10 is perfectly fine, I do not think we will ever produce sheets of size of a sail